adapters/rest-adapter-generator/src/main/java/uk/gov/justice/services/adapters/rest/generator/RestAdapterGenerator.java

```diff
@@ -103,18 +105,33 @@ private void writeToBasePackage(final TypeSpec typeSpec, final GeneratorConfig c
 
         final List<String> implementationNames = new ArrayList<>();
 
-        typeSpecs.stream().forEach(typeSpec -> {
-            try {
-                JavaFile.builder(configuration.getBasePackageName() + packageName, typeSpec)
-                        .build()
-                        .writeTo(configuration.getOutputDirectory());
-                implementationNames.add(typeSpec.name);
-            } catch (IOException e) {
-                throw new IllegalStateException(e);
-            }
-        });
+        typeSpecs.stream()
+                .forEach(typeSpec -> {
+                    try {
+                        if (!classExists(configuration, typeSpec)) {
+                            JavaFile.builder(configuration.getBasePackageName() + packageName, typeSpec)
+                                    .build()
+                                    .writeTo(configuration.getOutputDirectory());
+                        }
+
+                        implementationNames.add(typeSpec.name);
+                    } catch (IOException e) {
+                        throw new IllegalStateException(e);
+                    }
+                });
 
         return implementationNames;
     }
 
+    private boolean classExists(final GeneratorConfig configuration, final TypeSpec typeSpec) {
```

Comment:
Maybe the logic could be inverted for this method and rename it classDoesNotExist?